mbed-client-pal/Source/Port/Reference-Impl/OS_Specific/ZephyrOS/RTOS/pal_plat_rtos.c

```diff
@@ -290,23 +291,23 @@ palStatus_t pal_plat_osTimerStart(palTimerID_t timerID, uint32_t millisec)
 
     k_mutex_lock(&timer_data->lock, K_FOREVER);
 
+    int ret = 0;
+
     timer_data->timeout = (uptime + millisec);
     if (timer_data->period) {
         timer_data->period = millisec;
     }
 
-    if (!k_work_cancel_delayable(&timer_data->work)) {
+    ret = k_work_cancel_delayable(&timer_data->dwork);
```

Comment:
can we cancel delayable sync instead?
I don't think PAL timer start or stop would be executed from atomic context

Reply:
Hi @marcuschangarm , is it the right assumption?

Reply:
Having it sync is a trip to deadlock with current implementation. We are locking access to timer data. If I make this sync then the work_fn must be finished so that function could return. This will never happen because handler would not be able to lock the mutex (at the beggining of work_fn)

Reply:
I don't see anywhere in the code where start/stop is called from interrupt context. It might be called from the work queue itself though.

Reply:
Thanks.

> It might be called from the work queue itself though

I think that we can handle this case.